packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-cover.php

```diff
@@ -0,0 +1,217 @@
+		// Get original wrapper classes from block content.
+		$original_wrapper_classname = ( new Dom_Document_Helper( $block_content ) )->get_attribute_value_by_tag_name( 'div', 'class' ) ?? '';
+
+		// Get background color information.
+		$background_color = $this->get_background_color( $block_attrs, $rendering_context );
+
+		// Get block styles using the Styles_Helper.
+		$block_styles = Styles_Helper::get_block_styles( $block_attrs, $rendering_context, array( 'padding', 'border', 'background-color' ) );
+		$block_styles = Styles_Helper::extend_block_styles(
+			$block_styles,
+			array(
+				'width'           => '100%',
+				'border-collapse' => 'collapse',
+				'text-align'      => 'center',
+			)
+		);
+
+		// Add background image to table styles if present.
+		if ( ! empty( $background_image ) ) {
+			$block_styles = Styles_Helper::extend_block_styles(
+				$block_styles,
+				array(
+					'background-image'    => 'url("' . esc_url( $background_image ) . '")',
+					'background-size'     => 'cover',
+					'background-position' => 'center',
+					'background-repeat'   => 'no-repeat',
+				)
+			);
+		} elseif ( ! empty( $background_color ) ) {
+			// If no background image but there's a background color, use it.
+			$block_styles = Styles_Helper::extend_block_styles(
+				$block_styles,
+				array(
+					'background-color' => $background_color,
+				)
+			);
+		}
+
+		// Apply class and style attributes to the wrapper table.
+		$table_attrs = array(
+			'class' => 'email-block-cover ' . esc_attr( $original_wrapper_classname ),
+			'style' => $block_styles['css'],
+			'align' => 'center',
+			'width' => '100%',
+		);
+
+		// Build the cover content without background (background is now on the table).
+		$cover_content = $this->build_cover_content( $inner_content );
+
+		// Build individual table cell.
+		$cell_attrs = array(
+			'valign' => 'middle',
+			'align'  => 'center',
+		);
+
+		$cell = Table_Wrapper_Helper::render_table_cell( $cover_content, $cell_attrs );
+
+		// Use render_cell = false to avoid wrapping in an extra <td>.
+		return Table_Wrapper_Helper::render_table_wrapper( $cell, $table_attrs, array(), array(), false );
+	}
+
+	/**
+	 * Extract background image from block attributes or HTML content.
+	 *
+	 * @param array  $block_attrs Block attributes.
+	 * @param string $block_content Original block content.
+	 * @return string Background image URL or empty string.
+	 */
+	private function extract_background_image( array $block_attrs, string $block_content ): string {
+		// First check block attributes for URL.
+		if ( ! empty( $block_attrs['url'] ) ) {
+			return esc_url( $block_attrs['url'] );
+		}
+
+		// Fallback: use regex to find background image src directly.
+		if ( preg_match( '/<img[^>]*class="[^"]*wp-block-cover__image-background[^"]*"[^>]*src="([^"]+)"/', $block_content, $matches ) ) {
```

Comment:
This seems like a nice candidate for using WordPress HTML API. Have you checked if it is suitable for this use case?

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/60837/commits/c6a9e9f00c75e4f748636623ea96b80cf338d0b1 thanks for the suggestion!